librz/core/cmd_analysis.c

```diff
@@ -5082,6 +5083,46 @@ RZ_API int rz_core_esil_step_back(RzCore *core) {
 	return -1;
 }
 
+RZ_API bool rz_core_esil_continue_back(RzCore *core) {
+	rz_return_val_if_fail (core->analysis->esil && core->analysis->esil->trace, false);
+	RzAnalysisEsil *esil = core->analysis->esil;
+	if (esil->trace->idx == 0) {
+		return true;
+	}
+
+	RzRegItem *ripc = rz_reg_get (esil->analysis->reg, "PC", -1);
+	RzVector *vreg = ht_up_find (esil->trace->registers, ripc->offset | (ripc->arena << 16), NULL);
+	if (!vreg) {
+		RZ_LOG_ERROR ("failed to find PC change vector\n");
+		return false;
+	}
+
+	// Search for the nearest breakpoint in the tracepoints before the current position
+	bool bp_found = false;
+	int idx = 0;
+	RzAnalysisEsilRegChange *reg;
+	rz_vector_foreach_prev (vreg, reg) {
+		if (reg->idx >= esil->trace->idx) {
+			continue;
+		}
+		bp_found = rz_bp_get_in (core->dbg->bp, reg->data, RZ_BP_PROT_EXEC) != NULL;
+		if (bp_found) {
+			idx = reg->idx;
+			eprintf ("hit breakpoint at: 0x%" PFMT64x " idx: %d\n", reg->data, reg->idx);
+			break;
+		}
+	}
+
+	// Return to the nearest breakpoint or jump back to the first index if a breakpoint wasn't found
+	if (bp_found) {
```

Comment:
is this really necessary?
i think it can be just `rz_analysis_esil_trace_restore (esil, idx)`. If `bp_found=false` it means that we never entered the `if (bp_found)` at line 5109, so `idx` was never set. And the default value of `idx` is already 0.